### DISCUSSION TITLE
Use FlowAppsMigrator to duplicate a SharePoint with flows - What is the procedure

### DISCUSSION THREAD

**Discussion by sjorz70** [Zerg00s/FlowPowerAppsMigrator#70] [Q&A]
I am checking wit our IT to duplicate our current sharepoint with a lot of lists, lookupfields, flows and powerapps. IT wants to help us with ShareGate, but in my understanding with ShareGate it is not possible to migrate the apps and flows correctly (if not true, please let me know).

What are the instructions to use this tool when you want to duplicate the sharepoint and the flows? It should behave like a copy, so the flows must be renamed and imported as new flows, attached to the imported lists.

Are there any specific instructions for this or do I use the exact same for importing into a new tenant?

**Comment by Zerg00s** [comment]
Hi @sjorz70,

It's great that your company has access to ShareGate. In this case, the first step is to copy your existing site using it.
The next step is to export all your Power Apps and Flows on disk and place them into the \src subfolder in the Migrator folder.
Then, run the Migrator script and point it to the source and the destination sites. Since you are using ShareGate, don't let the Migrator migrate SharePoint lists and libraries. 
Lastly, when the Migrator finishes running, all your apps and flows will be converted and placed in the \dist folder. Manually deploy them. One-by-one.

**Note**
If possible, use the App registration. It's much more reliable as opposed to not using it.

**Comment by sjorz70** [comment]
Thank you Dennis!

Currently, we have an issue with the ShareGate license. Is the tool also reliable in copying the lists and files in case we cannot renew the license in time?

Kind regards,
Sjoerd de Vries

Verzonden vanuit Outlook voor iOS<https://aka.ms/o0ukef>
________________________________
Van: Denis Molodtsov ***@***.***>
Verzonden: Wednesday, November 29, 2023 12:51:11 AM
Aan: Zerg00s/FlowPowerAppsMigrator ***@***.***>
CC: Vries, Sjoerd de ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [Zerg00s/FlowPowerAppsMigrator] Use FlowAppsMigrator to duplicate a SharePoint with flows - What is the procedure (Discussion #70)

You don't often get email from ***@***.*** Learn why this is important<https://aka.ms/LearnAboutSenderIdentification>
Caution: This e-mail originated from outside of Philips, be careful for phishing.


Hi @sjorz70<https://github.com/sjorz70>,

It's great that your company has access to ShareGate. In this case, the first step is to copy your existing site using it.
The next step is to export all your Power Apps and Flows on disk and place them into the \src subfolder in the Migrator folder.
Then, run the Migrator script and point it to the source and the destination sites. Since you are using ShareGate, don't let the Migrator migrate SharePoint lists and libraries.
Lastly, when the Migrator finishes running, all your apps and flows will be converted and placed in the \dist folder. Manually deploy them. One-by-one.

Note
If possible, use the App registration. It's much more reliable as opposed to not using it.

—
Reply to this email directly, view it on GitHub<https://github.com/Zerg00s/FlowPowerAppsMigrator/discussions/70#discussioncomment-7698194>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BEJ222QIEDHSPH6VZBSXJGLYGZ2G7AVCNFSM6AAAAAA75WJT7WVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TMOJYGE4TI>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

________________________________
The information contained in this message may be confidential and legally protected under applicable law. The message is intended solely for the addressee(s). If you are not the intended recipient, you are hereby notified that any use, forwarding, dissemination, or reproduction of this message is strictly prohibited and may be unlawful. If you are not the intended recipient, please contact the sender by return e-mail and destroy all copies of the original message.

**Reply by Zerg00s** [reply]
The Migrator is using the PnP Powershell module for exporting the site schema from a source site and applying it at the destination site. This module is not reliable, but it's safe to try and it's  the best free tool available today. 

The Migrator does not move files, pages and list items. Lists and libraries a migrated with no content. Moving the content can be done manually or with the help of other tools like ShareGate.

**Comment by sjorz70** [comment]
Thank you! It makes sense now.

I tried and it does not seem to work. Without moving the lists I can convert the files. That would be already helpful.

I will ask IT to proceed checking ShareGate availability and help me with moving the files and lists first.

If it works, I certainly will use the GiveCoffee option from my personal savings account.

With Kind regards,

Sjoerd de Vries
EI IE S&SD - Global Learning Services
